File: admin/src/types/feedback.ts
/*
项目名称：JeriBlog
文件名称：feedback.ts
创建时间：2026-04-16 15:08:10

系统用户：Jerion
作　　者：Jerion
联系邮箱：416685476@qq.com
功能描述：类型定义 - feedback类型
*/

// 举报类型
export type ReportType = 'copyright' | 'inappropriate' | 'summary' | 'suggestion'

// 反馈状态
export type FeedbackStatus = 'pending' | 'resolved' | 'closed'

// 反馈内容结构
export interface FeedbackContent {
  description: string
  reason?: string
  attachmentFiles?: string[]
}

// 反馈对象
export interface Feedback {
  id: number
  ticket_no: string
  report_url: string
  report_type: ReportType
  form_content: FeedbackContent
  email: string
  status: FeedbackStatus
  admin_reply: string
  reply_time?: string
  user_agent: string
  ip: string
  feedback_time: string
}

// 反馈列表数据
export interface FeedbackListData {
  list: Feedback[]
  total: number
  page: number
  page_size: number
}

// 更新反馈请求
export interface FeedbackUpdateRequest {
  status: FeedbackStatus
  admin_reply?: string
}

File: server/internal/model/feedback.go
package model

import "time"

// Feedback 投诉举报模型
type Feedback struct {
	ID           uint       `gorm:"primarykey" json:"id"`
	TicketNo     string     `gorm:"type:varchar(50);unique;not null" json:"ticket_no"` // 工单号
	ReportUrl    string     `gorm:"type:varchar(500);not null" json:"report_url"`      // 举报URL
	ReportType   string     `gorm:"type:varchar(50);not null" json:"report_type"`      // 举报类型
	FormContent  string     `gorm:"type:text;not null" json:"form_content"`            // 表单内容(JSON格式)
	Email        string     `gorm:"type:varchar(255)" json:"email"`                    // 联系邮箱(可选)
	Status       string     `gorm:"type:varchar(20);default:'pending'" json:"status"`  // 状态
	AdminReply   string     `gorm:"type:text" json:"admin_reply"`                      // 管理员回复
	ReplyTime    *time.Time `json:"reply_time"`                                        // 回复时间
	UserAgent    string     `gorm:"type:varchar(500)" json:"user_agent"`               // 用户代理
	IP           string     `gorm:"type:varchar(45)" json:"ip"`                        // IP地址
	FeedbackTime time.Time  `gorm:"not null" json:"feedback_time"`                     // 反馈时间
	CreatedAt    time.Time  `json:"created_at"`                                        // 创建时间
	UpdatedAt    time.Time  `json:"updated_at"`                                        // 更新时间
}
